### PR TITLE
fixes #1145 - Scale app after killing tasks without scaling

### LINKED ISSUE #1145: 0.8.0: 'kill task' does not allow new task to take its place
In 0.7.6, if you killed an task with the UI it was replaced by another task. This was useful whenever you had to restart a specific task while 'kill & scale' actually removed the task and scaled down. However in 0.8.0, if you kill a task, it is just being killed and the number of running is decremented forever. In order to respawn a new task, you have to 'scale down' to the new decremented value and scale back up to the original value.

Reproduction:
1) Start a docker app with 5 instances
2) When all 5 are running, open up the UI, click the app, and check one of the tasks and press the Kill button
3) Observe the task being killed. 
4) Exit out the modal and refresh the page
5) Observe that the task count remains 4, even though it should go back to 5

Workaround:
1) Whenever it is stuck at 4 value, go back to the modal, click 'scale' and scale to 4 first. 
2) Click scale again, and scale back up to 5

### PR DESCRIPTION
**Pull request by drexin**
- added force parameter to ScaleApp message to prevent races between
  the finishing of KillTasks and the execution of ScaleApp

### CODE REVIEW

**Comment by sttts** [comment]
Can you put some explanation why we need this force attribute into the commit msg? Without it's hard to follow.

**Comment by drexin** [comment]
Good idea, thanks.

**Comment by sttts** [comment]
Now I understand :-)

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala [hunk 1]
```diff
@@ -300,7 +303,7 @@ object MarathonSchedulerActor {
 
   case object ReconcileHealthChecks
 
-  case class ScaleApp(appId: PathId) extends Command {
+  case class ScaleApp(appId: PathId, force: Boolean = false) extends Command {
```
Comment (ConnorDoyle) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: Please document the semantics of the `force` parameter here.  The other commands are also lacking but let's ++docblocks going forward.

Comment (ConnorDoyle) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: Also, since there are only two use-sites total I'm not sure the default value buys us much.  On the other hand, it makes the use site where the default is not overridden harder to understand.

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala [hunk 2]
```diff
@@ -163,7 +163,10 @@ class MarathonSchedulerActor(
               .getOrElse(Future.successful(()))
           } yield ()
         }
-        else promise.future
+        else promise.future andThen {
+          case _ =>
+            self ! ScaleApp(appId, force = true)
```
Comment (ConnorDoyle) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: It makes sense to do this here.  However, I'm guessing it was exposed because we recently increased the default task reconciliation interval.  Created issue #1177 for decoupling the frequency of global scale operations from the task reconciliation frequency.
